### PR TITLE
Issue #74: update PatchFilter to use jgit to parse patch file for line strategy

### PR DESCRIPTION
**Pull request by HuGanghui**
Issue #74: update PatchFilter to use jgit to parse patch file for line strategy

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilter.java [hunk 1]
```diff
@@ -98,74 +116,59 @@ public boolean accept(AuditEvent event) {
...
-        for (int i = 0; i < patch.getDeltas().size(); i++) {
-            final Chunk targetChunk = patch.getDeltas().get(i).getTarget();
-            final List<Integer> lineRange = new ArrayList<>();
-            lineRange.add(targetChunk.getPosition());
-            lineRange.add(targetChunk.getPosition() + targetChunk.getLines().size());
-            lineRangeList.add(lineRange);
+        if (!"RENAME".equals(fileHeader.getChangeType().name())) {
+            for (HunkHeader hunkHeader : fileHeader.getHunks()) {
+                final EditList edits = hunkHeader.toEditList();
+                for (int i = 0; i < edits.size(); i++) {
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilter.java:152]: can we avoid `i` at all ? `for (Edit edit: edits)`

File: src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterTest.java [hunk 2]
```diff
@@ -57,6 +57,7 @@ protected String getPackageLocation() {
     }
 
     @Test
+    @Ignore
```
[the remark this answers is not in the document]
Comment (HuGanghui) [reply src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterTest.java]: done

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilter.java (hunk 1 above)
Comment (HuGanghui) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilter.java]: done

**Comment by rdiachenko** [comment]
Merged in scope of https://github.com/checkstyle/patch-filters/pull/86

**Pull request closed without merging**